.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,35 @@
+# Description
+
+Please include a summary of the change or the fix you've made. Please also include relevant motivation and context. List any dependencies that are required for this change.
+
+Fixes # (issue)
+
+## Type of change
+
+Please select atleast one
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
```

Comment:
this one can be difficult for new contributors, as they might not understand how this could impact existing users. not saying we shouldn't have it, but wanted to call it out as I've personally been in that situation myself

Reply:
thats a good callout. I can see myself being confused about this too.  We could add a `I'm not sure` option in it. wdyt?

Reply:
I'd personally lean towards leaving it out but curious what others think

Reply:
I think we can tweak the language here. Maybe `Breaking API Change`?

Reply:
Can you automate this as well? In kotlin we've been automating this for a few months, at a binary compatibility level.

Reply:
I can look into automating this. As suggested for adding linter to CI, i would suggest tracking this with a different PR.